docs/source/index.rst

```diff
@@ -59,6 +59,7 @@ Features described in this documentation are classified by release status:
    compile/deep-dive
    compile/cudagraph_trees
    compile/performance-dashboard
+   compile/profiling_torch_compile
```

Comment:
one small nit - want to link to this directly above the gpu profiling section ? note that the references should be fixed here: https://github.com/pytorch/pytorch/pull/103014/ and https://github.com/pytorch/pytorch/pull/102407... up to you

also when landing you can just wait for docs to build then merge -f